util/argo/resource_tracking_test.go

```diff
@@ -204,6 +204,50 @@ func TestResourceIdNormalizer_Normalize(t *testing.T) {
 	assert.False(t, hasOldLabel)
 }
 
+func TestResourceIdNormalizer_NormalizeCRD(t *testing.T) {
+	rt := NewResourceTracking()
+
+	// live object is a CRD resource
+	liveObj := &unstructured.Unstructured{
+		Object: map[string]any{
+			"apiVersion": "apiextensions.k8s.io/v1",
+			"kind":       "CustomResourceDefinition",
+			"metadata": map[string]any{
+				"name": "crontabs.stable.example.com",
+				"labels": map[string]any{
+					common.LabelKeyAppInstance: "my-app",
+				},
+			},
+			"spec": map[string]any{
+				"group": "stable.example.com",
+				"scope": "Namespaced",
+			},
+		},
+	}
+
+	// config object is a CRD resource
+	configObj := &unstructured.Unstructured{
```

Comment:
```suggestion
	configObj := &unstructured.Unstructured{
```

what is configObj representing here? The CRD present in git?

Reply:
@nitishfy that's right, the `configObj` is a manifest that the repo-server has extracted from Git (or Helm) and the application-controller in turn would like to apply to the cluster. The `liveObj` is the actual object found in the cluster.